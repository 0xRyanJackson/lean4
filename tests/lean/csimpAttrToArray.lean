set_option trace.compiler.ir.init true
def f (as : List Nat) : Array Nat :=
  as.toArray
