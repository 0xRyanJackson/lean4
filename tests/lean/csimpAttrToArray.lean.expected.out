
[init]
def f (x_1 : obj) : obj :=
  let x_2 : obj := List.redLength._rarg x_1;
  let x_3 : obj := Array.mkEmpty â—¾ x_2;
  let x_4 : obj := List.toArrayAux._rarg x_1 x_3;
  ret x_4